controller.go

```diff
@@ -119,7 +120,15 @@ type driverData struct {
 	capability driverapi.Capability
 }
 
+type ipamData struct {
+	config    ipamapi.Config
+	allocator ipamapi.Allocator
+	// default address spaces are provided by ipam driver at registration time
+	defaultLocalAddressSpace, defaultGlobalAddressSpace string
```

Comment:
@mapuri
Sorry for the late response:
It is not a requirement that a remote ipam driver must serve both local and global scope. As long as it is used in conjunction with the appropriate network driver.

It is instead a requirement for the Default IPAM to serve both local and global scoped net drivers simply because being the default  builtin ipam driver, it has to serve both the inbuilt bridge and overlay drivers.